src/omv/modules/py_tf.c

```diff
@@ -803,31 +708,384 @@ STATIC void py_tf_regression_input_callback(void *callback_data,
+
+    size_t len = list_size(&nms_bounding_boxes);
+    for (size_t i = 0; i < len; i++) {
+        py_tf_model_output_bounding_box_lnk_data_t lnk_data;
+        list_pop_front(&nms_bounding_boxes, &lnk_data);
+        py_tf_classification_obj_t *o = m_new_obj(py_tf_classification_obj_t);
+        o->base.type = &py_tf_classification_type;
+        o->x = mp_obj_new_int(fast_floorf(lnk_data.rect.x * scale) + x_offset);
+        o->y = mp_obj_new_int(fast_floorf(lnk_data.rect.y * scale) + y_offset);
+        o->w = mp_obj_new_int(fast_floorf(lnk_data.rect.w * scale));
+        o->h = mp_obj_new_int(fast_floorf(lnk_data.rect.h * scale));
+        o->output = mp_obj_new_float(lnk_data.score);
+        mp_obj_list_append(((mp_obj_list_t *) list)->items[lnk_data.label_index], o);
+    }
+
+    return list;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_KW(py_tf_model_output_get_bounding_boxes_obj, 1, py_tf_model_output_get_bounding_boxes);
+
+STATIC const mp_rom_map_elem_t py_tf_model_output_locals_table[] = {
+    { MP_ROM_QSTR(MP_QSTR_add_bounding_box),    MP_ROM_PTR(&py_tf_model_output_add_bounding_box_obj) },
+    { MP_ROM_QSTR(MP_QSTR_get_bounding_boxes),  MP_ROM_PTR(&py_tf_model_output_get_bounding_boxes_obj) },
+};
+
+STATIC MP_DEFINE_CONST_DICT(py_tf_model_output_locals_dict, py_tf_model_output_locals_table);
+
+STATIC MP_DEFINE_CONST_OBJ_TYPE(
+    py_tf_model_output_type,
+    MP_QSTR_tf_model_output,
+    MP_TYPE_FLAG_NONE,
+    attr, py_tf_model_output_attr,
+    subscr, py_tf_model_output_subscr,
+    locals_dict, &py_tf_model_output_locals_dict
+    );
+
+typedef struct py_tf_invoke_callback_data {
+    rectangle_t *roi;
+    mp_obj_t callback;
+    mp_obj_t *out;
+} py_tf_invoke_callback_data_t;
+
+STATIC void py_tf_invoke_output_callback(void *callback_data,
+                                         void *model_output,
+                                         libtf_parameters_t *params) {
+    py_tf_invoke_callback_data_t *arg = (py_tf_invoke_callback_data_t *) callback_data;
+    py_tf_model_output_obj_t *o = m_new_obj(py_tf_model_output_obj_t);
+    o->base.type = &py_tf_model_output_type;
+    o->roi = arg->roi;
+    o->model_output = model_output;
+    o->params = params;
+    o->output_size = params->output_height * params->output_width * params->output_channels;
+    o->x = mp_obj_new_int(arg->roi->x);
+    o->y = mp_obj_new_int(arg->roi->y);
+    o->w = mp_obj_new_int(arg->roi->w);
+    o->h = mp_obj_new_int(arg->roi->h);
+    o->rect = mp_obj_new_tuple(4, (mp_obj_t []) {o->x, o->y, o->w, o->h});
+    o->input_height = mp_obj_new_int(params->input_height);
+    o->input_width = mp_obj_new_int(params->input_width);
+    o->input_channels = mp_obj_new_int(params->input_channels);
+    o->output_height = mp_obj_new_int(params->output_height);
+    o->output_width = mp_obj_new_int(params->output_width);
+    o->output_channels = mp_obj_new_int(params->output_channels);
```

Comment:
input/output_shape tuples  ?

Reply:
I will do this after clarification: https://github.com/openmv/openmv/pull/2227#discussion_r1632393552

Reply:
Done